### PR TITLE
CS/SQL: refactor query using ` IN ()`

### PR DESCRIPTION
**Pull request by jrfnl**
## Summary
This PR can be summarized in the following changelog entry:
* Security hardening

## Relevant technical choices:

Queries with a variable number of replacements can be prepared as well.

The placeholders are dynamically created using an `implode( ',', array_fill( 0, $count, '%s' ) )` and the complete set of replacements need to be passed as an array in the second parameter of `prepare()`.

Includes:
* Prepare late, i.e. wrap the call to `prepare()` within the query call.
* Remove a `private` method which previously did some work for the `IN` clause.

## Test instructions
* Verify that the sitemap still displays all the items expected.
* With a debug tool - check that the SQL query created and run is correct.

### CODE REVIEW

File: classes/class-sitemap.php [hunk 1]
```diff
@@ -196,28 +196,29 @@ private function get_stylesheet_line() {
...
 
-		// @codingStandardsIgnoreEnd
+		// Get posts for the last two days only, credit to Alex Moss for this code.
+		$items = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT ID, post_content, post_name, post_author, post_parent, post_date_gmt, post_date, post_date_gmt, post_title, post_type
+				FROM {$wpdb->posts}
+				WHERE post_status='publish'
+					AND (DATEDIFF(CURDATE(), post_date_gmt)<=2)
+					AND post_type IN (" . implode( ',', array_fill( 0, count( $post_types ), '%s' ) ) . ')
```
Comment (jdevalk) [inline classes/class-sitemap.php]: I think this whole `implode` etc. is kinda hard to follow when reading, can't we do this outside of the `prepare`?

File: classes/class-sitemap.php [hunk 2]
```diff
@@ -196,28 +196,29 @@ private function get_stylesheet_line() {
...
-			 WHERE post_status=%s
-			 AND (DATEDIFF(CURDATE(), post_date_gmt)<=2)
-			 AND post_type IN ({$post_types})
-			 ORDER BY post_date_gmt DESC
-			 LIMIT 0, {$limit}
-		 ";
-
-		$items = $wpdb->get_results( $wpdb->prepare( $sql_query, 'publish' ) );
+		$replacements   = $post_types;
+		$replacements[] = max( 1, min( 1000, $limit ) );
```
Comment (jdevalk) [inline classes/class-sitemap.php]: What is this for? A comment on what this means would be useful for me :)

File: classes/class-sitemap.php (hunk 1 above)
Comment (jrfnl) [reply classes/class-sitemap.php]: > can't we do this outside of the prepare?

@jdevalk Not really. If you would, you would still need to pass a variable into the query which will get flagged as needing preparation, but can't use prepare, as the variable being passed in is the string with the placeholders to do the replacements with....
Creating the query with a `sprintf()` before doing the prepare won't work either as then we'd get tripped up by the placeholder in the `limit` clause.

File: classes/class-sitemap.php (hunk 2 above)
Comment (jrfnl) [reply classes/class-sitemap.php]: @joost Agreed, except I only moved that code. I didn't write nor change it.

The way I read it:
* `$limit` is an arbitrary number;
* the target number should be between 1 and 1000;
* the snippet makes sure between those three numbers - `1`, `1000`, `$limit` - that the highest number is selected, but no higher than 1000.
